### DISCUSSION TITLE
Socket id generating .

### DISCUSSION THREAD

**Discussion by AthulKkumar** [socketio/socket.io#4820] [Q&A]
Socket.IO is creating 2 or more socket IDS every time a new user joins a room. 

This is  from socket.js 

`export const initSocket = async () => {
  const options = {
    "force new connection": true,
    reconnectionAttempts: "Infinity",
    timeout: 10000,
    transports: ["websocket"],
  };

  return io(process.env.REACT_APP_BACKEND_URL, options);
};`

This the component code :
`
import { initSocket } from "../socket";

....
const socketRef = useRef()

useEffect(( ) =>{
const init = async () => {
      socketRef.current = await initSocket();
}
 init()

}, [])
`

**Comment by darrachequesne** [comment]
Hi! Did you check our React guide [here](https://socket.io/how-to/use-with-react)?
